src/core/Akka.TestKit/Internal/BufferQueue.cs

```diff
@@ -33,20 +35,20 @@ public class BlockingQueue<T>
         /// Adds the specified item to the end of the queue.
         /// </summary>
         /// <param name="item">The item to add to the queue.</param>
-        public void Enqueue(T item)
-        {
-            if (!_collection.TryAdd(new Positioned(item)))
+        public async ValueTask Enqueue(T item)
+        { 
+            if (!await _collection.SendAsync(new Positioned(item)))
                 throw new InvalidOperationException("Failed to enqueue item into the queue.");
         }
 
         /// <summary>
         /// Adds the specified item to the front of the queue. 
         /// </summary>
         /// <param name="item">The item to add to the queue.</param>
-        public void AddFirst(T item)
+        public async ValueTask AddFirst(T item)
         {
-            if(!_collection.TryAdd(new Positioned(item, first:true)))
-                throw new InvalidOperationException("Failed to enqueue item into the head of the queue.");
+           if (!await _collection.SendAsync(new Positioned(item, first: true)))
```

Comment:
This will fail. This line is supposed to insert the item at the front of the queue, not at the back.